### PR TITLE
fix: fetching the plugin form config for queries if datasource is not viewable

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

> Fetching the plugin form config for queries if datasource is not viewable.

> TL;DR: Change is being done for GAC. Shouldn't impact users.

Fixes [#18372](https://github.com/appsmithorg/appsmith/issues/18372)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/entities/Engine/AppEditorEngine.ts [hunk 1]
```diff
@@ -172,10 +172,8 @@ export default class AppEditorEngine extends AppEngine {
   }
 
   public *loadAppEntities(toLoadPageId: string, applicationId: string): any {
-    yield all([
-      call(this.loadPageThemesAndActions, toLoadPageId, applicationId),
-      call(this.loadPluginsAndDatasources),
-    ]);
+    yield call(this.loadPageThemesAndActions, toLoadPageId, applicationId);
+    yield call(this.loadPluginsAndDatasources);
```
Comment (berzerkeer) [inline app/client/src/entities/Engine/AppEditorEngine.ts:176]: Is this dependent on the prev ? hence we removed fetching these resources in parallel ?

Comment (ankitakinger) [reply app/client/src/entities/Engine/AppEditorEngine.ts:176]: yes

**Review by berzerkeer — APPROVED** [review]

**Pull request merged**